Extension/src/LanguageServer/client.ts

```diff
@@ -2724,15 +2728,22 @@ export class DefaultClient implements Client {
 
         showConfigStatus = showConfigStatus || (configurationNotSet &&
             !!compilerDefaults && !compilerDefaults.trustedCompilerFound && trustedCompilerPaths && (trustedCompilerPaths.length !== 1 || trustedCompilerPaths[0] !== ""));
+        if (showConfigStatus) {
+            this.showConfigureIntelliSenseButton = true;
+        } else {
+            this.showConfigureIntelliSenseButton = false;
+        }
+        if (!configProviderNotSet) {
```

Comment:
Can this block of `if/else` be changed to something like:

```ts
let configType: ConfigurationType = none;
if (condition a) {
      configType = ConfigProvider;
} else if (condition b) {
    configType = CompileCommands;
} else if (......) {
    // etc.
}

// Call ShowConfigureIntelliSenseButton after evaluting the value for ConfigurationType
ui.ShowConfigureIntelliSenseButton(this.showConfigureIntelliSenseButton, this, configType, "handleConfig");
```

Reply:
sure